### PR TITLE
i#2796: test aflags are restored after clean call on aarch64

### PR DESCRIPTION
**Pull request by c-rhodes**
This commit implements codegen_aflags_clobber on aarch64, using the msr
instruction to set the NZCV system register. This will also be used by
the inline.dll.c tests, once the NZCV flags are restored there.

The x86 implementation of codegen_aflags_clobber has also been moved to
cleancall-opt-shared.h.

Issue: #2796

### CODE REVIEW

**Comment by fhahn** [comment]
run aarch64 tests
run aarch32 tests

**Comment by fhahn** [comment]
run aarch32 tests

File: suite/tests/client-interface/cleancall-opt-shared.h [hunk 1]
```diff
@@ -781,3 +781,43 @@ codegen_bbcount(void *dc)
     codegen_epilogue(dc, ilist);
     return ilist;
 }
+
+/* Clobber aflags.  Inliner must ensure they are restored.
```
Comment (fhahn) [inline suite/tests/client-interface/cleancall-opt-shared.h]: Please update the comment: Clean call optimizations must ensure they are restored.

**Comment by fhahn** [comment]
run aarch64 tests

**Comment by c-rhodes** [comment]
@fhahn thanks

**Pull request merged**